spec/controllers/announcements_controller_spec.rb

```diff
@@ -129,5 +149,8 @@
       end
       it_behaves_like 'access denied'
     end
+    context 'GET hide' do
```

Comment:
Gotcha. For this issue let's include separate tests for all roles and we'll refactor in #1390. I'm still looking into the `ability.rb` thing, it might actually be fundamentally broken :stuck_out_tongue:.